test-common/uk/gov/hmrc/alcoholdutyaccount/common/TestData.scala

```diff
@@ -224,13 +234,299 @@ trait TestData extends ModelGenerators {
+            FinancialTransactionItem(
+              subItem = "000",
+              dueDate = Some(ReturnPeriod.fromPeriodKeyOrThrow(periodKey).dueDate()),
+              amount = BigDecimal("-5000")
+            )
+          )
+        ),
+        FinancialTransaction(
+          sapDocumentNumber = sapDocumentNumberGen.sample.get,
+          periodKey = None,
+          chargeReference = None,
+          originalAmount = BigDecimal("-2000"),
+          mainTransaction = TransactionType.toMainTransactionType(TransactionType.PaymentOnAccount),
+          subTransaction = "6132",
+          outstandingAmount = Some(BigDecimal("-2000")),
+          items = Seq(
+            FinancialTransactionItem(
+              subItem = "000",
+              dueDate = Some(ReturnPeriod.fromPeriodKeyOrThrow(periodKey).dueDate()),
+              amount = BigDecimal("-2000")
+            )
+          )
+        )
+      )
+    )
+
+  val onePartiallyPaidReturnLineItemAndOnePartiallyAllocatedPaymentOnAccount: FinancialTransactionDocument =
+    FinancialTransactionDocument(
+      financialTransactions = Seq(
+        FinancialTransaction(
+          sapDocumentNumber = sapDocumentNumberGen.sample.get,
+          periodKey = Some(periodKey),
+          chargeReference = Some(chargeReferenceGen.sample.get),
+          originalAmount = BigDecimal("9000"),
+          mainTransaction = TransactionType.toMainTransactionType(TransactionType.Return),
+          subTransaction = "6132",
+          outstandingAmount = Some(BigDecimal("5000")),
+          items = Seq(
+            FinancialTransactionItem(
+              subItem = "000",
+              dueDate = Some(ReturnPeriod.fromPeriodKeyOrThrow(periodKey).dueDate()),
+              amount = BigDecimal("5000")
+            ),
+            FinancialTransactionItem(
+              subItem = "001",
+              dueDate = Some(ReturnPeriod.fromPeriodKeyOrThrow(periodKey).dueDate()),
+              amount = BigDecimal("4000")
+            )
+          )
+        ),
+        FinancialTransaction(
+          sapDocumentNumber = sapDocumentNumberGen.sample.get,
+          periodKey = None,
+          chargeReference = None,
+          originalAmount = BigDecimal("-2000"),
+          mainTransaction = TransactionType.toMainTransactionType(TransactionType.PaymentOnAccount),
+          subTransaction = "6132",
+          outstandingAmount = Some(BigDecimal("-2000")),
+          items = Seq(
+            FinancialTransactionItem(
+              subItem = "000",
+              dueDate = Some(ReturnPeriod.fromPeriodKeyOrThrow(periodKey).periodFromDate()),
+              amount = BigDecimal("-2000")
+            )
+          )
+        )
+      )
+    )
+
+  val singleFullyOutstandingLPI: FinancialTransactionDocument =
+    FinancialTransactionDocument(
+      financialTransactions = Seq(
+        FinancialTransaction(
+          sapDocumentNumber = sapDocumentNumberGen.sample.get,
+          periodKey = None,
```

Comment:
Like mentioned before, create template vals for Payment on account, LPI, RPI that have the correct fields and copy them to change specific values